### DISCUSSION TITLE
Firmware update on Windows 10 doens't work

### DISCUSSION THREAD

**Discussion by karmalvel** [solokeys/solo2#134] [Q&A]
Hi!

I got the email saying the new firmware is available and started the process to install rust and all that was informed on "https://solokeys.com/pages/firmware-update-for-windows". Now the problem is that when I try to do anything with the key, be it asking version or trying to update, I get the same error: "Error: Empty list of Solo 2". I have three usb-c -version keys and one usb-a and all have the same problem. I've tried to restart everything, reinstall everything and even tried with three computers, but all give the same error. Any and all help would be appreciated!

**Comment by needs-coffee** [comment]
@karmalvel  have you tried opening a cmd prompt as administrator and seeing if the solos are listed? Are you using a USB extension/hub? do the keys get recognised/usable by the computer when not updating? (i.e. with websites or in windows device manager)

EDIT
also you could try with the exe on the releases page and not using rust and see if this is the issue - https://github.com/solokeys/solo2-cli/releases/tag/v0.2.0

**Reply by ursa2** [reply]
I’m having the same problem with a Solo 1 USB-A that I just received.  I’m running Windows 10 Pro Version	10.0.19044 Build 19044. Under Devices -> Bluetooth & other d evices, Other devices, Solo 3.1.2. Under Device Manager, I do not find it listed under Security Devices, I only see Trusted Platform Module 2.0.  I tried installing cargo install solo2 and running as administrator.  I also tried as suggested solo2-v0.2.0-x86_64-pc-windows-msvc.exe (also with cmd at administrator) and the results are the same.  If I do a list from solo2 I get return to command prompt. If I do app admin version from either program, I get the Empty list of Solo 2 error. If I do update from either program I get Empty list of Solo 2 devices error.

The reason for the firmware update is that the device doesn’t work on Windows 10 without it. (see https://github.com/solokeys/solo2-cli/issues/35).  I tried and I get a red led and a timeout at webauthn.io.

**Reply by needs-coffee** [reply]
Hi @ursa2  - to be clear to do have a solo 1 or solo 2?

The device does not appear under `security devices` within device manager, it appears under `human interface devices` as `HID-Compliant fido`, under `Smart Cards` and `smart card readers` as Unknown, and comes under `USB controllers` as `USB composite Device`

If a solo2 you are looking for a USB device with VID 1209 and PID BEEE under Hardware ID's of a USB composite device. The easy method would be count the number of USB devices listed before and after connecting it.

Is it a USB port or Thunderbolt port?

**Reply by ursa2** [reply]
I ordered a Solo 1 USB-A Red/Black.  The package only says Solo USB-A.  Windows is calling it 3.1.2.  I'm using a USB 2 port.

I'm going to try the Solo on another computer tomorrow and see if that works.

**Reply by needs-coffee** [reply]
> I ordered a Solo 1 USB-A Red/Black.

then the solo2-cli wont work - you need the https://github.com/solokeys/solo1-cli to update this. This is the repo for the solo2 which is different. The latest [solo1](https://github.com/solokeys/solo1) firmware is 4.1.5

**Reply by JohnLGalt** [reply]
I just received V2 keys, LEs at that, a couple of days ago. I went through the process of setting everything up, got the VS part installed, Rust installed, and cargo install solo2 installed package 'solo2 v0.2.1', verified by running cargo install solo2 again and it stating it was already installed and thus Ignored it.

One key is the regular Type A key, the other is the Type-C Hacker. I've been interested in mainly setting up the Type A key for now to add it to my other keys for my accounts, but I've tried, just for S&G, both keys, connected to my system, They appear just fine in Device manager, as you listed above, in all three spots exactly as you state. However, running `solo2 app admin version` in a standard cmd window (or cmd instance in Terminal ,tried both) has it take a long time before returning `Error: Empty list of Solo 2`. I just tried in an admin-elevated cmd window and the command has not done anything thus far. It's been literally 5 minutes waiting for a response.

One thing to note - when I first set everything up, and then inserted the Type A key for the first time in a USB 3.0 slot, Windows wanted to reboot before it could use the key. I'm going to try to remove all  of those instances of the key from the system, reboot, and then insert the key and see if it reinstalls the drivers, and see if that fixes it.

Windows 11 22H2 (build 22621.674)
Windows Defender - Real-Time Protection and Tamper protection enabled, Controlled folder access disabled, Core Isolation (Memory Integrity) enabled

Anything else that you may think I should try to get this working?

EDIT: The elevated cmd window just returned `Error: Empty list of Solo 2` - as I expected it would. Gonna kill drivers and try again, see if it makes a difference.

**Reply by JohnLGalt** [reply]
After doing all of the above, same results, except for one thing - it is not taking as long.

I thought it might have been because my Titan Security keys were inserted, confusing the system, so I unplugged that, and then deleted all the drivers as mentioned above. With the Hacker Type-C the light stayed blue, but same Error. With the regular Type A, light was green, but same error also. Checking DM, I see all the same entries as before for all the devices.

Going to try one last thing, that is disconnecting every external USB hub that I have (been using the connections on the case, not the hubs), and try using the back I/O panel connectors to see if that makes any sore of difference, even though I know the case connectors are wired directly off on motherboard USB connectors.

Also to note:

On my rig, the entry under `Smart card readers` is `Microsoft Usbccid Smartcard Reader (WUDF)` and under `Smart cards` as `Unknown Smart Card`.

**Reply by ursa2** [reply]
Sorry have been away.  I got the Solo to work on two other computers with webauthn.io.  So, the problem with it not working is only on the first computer I tried.  Since it works, the problem if with only that one computer and I will try to figure out why.

Thank you for the help.
